tools/travis.ps1

```diff
@@ -225,12 +231,20 @@ elseif($Stage -eq 'Build')
         Remove-Item -force ${telemetrySemaphoreFilepath}
     }
 
-    $pesterPassThruObject = Start-PSPester @pesterParam
+    # Running tests which do not require sudo.
+    Start-PSPester @pesterParam
+
+    # Running tests, which require sudo.
+    pesterParam['Tag'] = @('RequireSudoOnUnix')
+    pesterParam['ExcludeTag'] = @()
+    pesterParam['Sudo'] = $true
+    pesterParam['OutputFile'] = $testResultsSudo
+    Start-PSPester @pesterParam
 
     # Determine whether the build passed
     try {
         # this throws if there was an error
-        Test-PSPesterResults -ResultObject $pesterPassThruObject
+        @($testResultsNoSudo, $testResultsSudo) | ForEach-Object { Test-PSPesterResults -TestResultsFile $_ }
```

Comment:
please don't change to the files, use the result objects.  The files can cause be partial, and cause bugs.

Reply:
So I should just keep the code as it was before with pesterPassThru objects, but just add the output files?